airflow/models/backfill.py

```diff
@@ -347,7 +358,7 @@ def _get_info_list(
     now = timezone.utcnow()
     dagrun_info_list = (x for x in infos if x.data_interval.end < now)
     if reverse:
-        dagrun_info_list = reversed([x for x in dag.iter_dagrun_infos_between(from_date, to_date)])
+        dagrun_info_list = reversed([x for x in infos if x.data_interval.end < now])
```

Comment:
```suggestion
        dagrun_info_list = reversed(dag_run_info_list)
```

The < now filter was laready applied 2 lines above